homeassistant/components/template/light.py

```diff
@@ -320,6 +408,13 @@ async def async_turn_on(self, **kwargs):
             self._white_value = kwargs[ATTR_WHITE_VALUE]
             optimistic_set = True
 
+        if self._effect_list_template is None and ATTR_EFFECT in kwargs:
+            _LOGGER.info(
+                "Optimistically setting effect list to %s", kwargs[ATTR_EFFECT]
+            )
```

Comment:
Instead of this:
- Make it mandatory to configure `CONF_EFFECT_LIST_TEMPLATE` and `CONF_EFFECT_ACTION` together by making them inclusive instead of optional (see suggestion above)
- Check that the effect in the service call is a member of the effect list